### PR TITLE
Simplifying a bit the data handling, now most of it is done through hooks

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: page/src/components/Filters/Filters.jsx [hunk 1]
```diff
@@ -39,15 +39,15 @@ const Filters = ({ query, callForPapers, closedCaptions, country, onChange, onCl
       </div>
 
       <div className='filtersItem'>
-        <label for='filter-closed-captions'>Closed Captions:</label>
+        <label htmlFor='filter-closed-captions'>Closed Captions:</label>
```
Comment (jespino) [inline page/src/components/Filters/Filters.jsx:42]: this file changes was fixing some warnings

File: page/src/components/MapView/MapView.jsx [hunk 2]
```diff
@@ -44,31 +42,12 @@ const MapView = ({year}) => {
     }, {})
   }, [events]);
 
-  const monthNames = [
```
Comment (jespino) [inline page/src/components/MapView/MapView.jsx:47]: I moved this kind of code to the utils file

File: page/src/main.jsx [hunk 3]
```diff
@@ -2,9 +2,6 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 
 import App from './App';
-import {getEventsByYear} from 'utils';
```
Comment (jespino) [inline page/src/main.jsx:5]: This is no longer use with this `window.dev_events` object

File: page/src/components/CalendarGrid/CalendarGrid.jsx [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
+import {useMemo} from 'react';
 
 import Calendar from 'components/Calendar/Calendar';
 
 import 'styles/CalendarGrid.css';
 
 const CalendarGrid = ({year}) => {
-  const [months, setMonths] = useState([]);
-
-  useMemo(() => {
```
Comment (jespino) [inline page/src/components/CalendarGrid/CalendarGrid.jsx:10]: This use of `useMemo` + the `useState` was weird, useMemo is for memoizing data, so I changed that.

File: page/src/components/Calendar/Calendar.jsx [hunk 5]
```diff
@@ -1,29 +1,24 @@
...
 
 const DaysName = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su'];
 
 const Calendar = ({year, month, days}) => {
-  const [daysEvents, setDaysEvents] = useState(null);
   const userDispatch = useCustomContext().userDispatch;
+  const yearEvents = useYearEvents()
+  const monthEvents = useMonthEvents(yearEvents, month)
 
-  useMemo(() => {
```
Comment (jespino) [inline page/src/components/Calendar/Calendar.jsx:16]: Same here, useMemo+useState is weird IMO.

File: page/src/components/Filters/Filters.jsx (hunk 1 above)
Comment (scraly) [reply page/src/components/Filters/Filters.jsx:42]: ah ah I wanted to do the same thing, thx :)

**Comment by jespino** [comment]
@scraly do you want me to split this PR in smaller ones to make it easier to merge?

**Comment by scraly** [comment]
I will try to take a look thie evening :)

**Comment by scraly** [comment]
As you fix warning and you are doing a refactor,
here is a warning message when we click on the list view:
image

**Comment by jespino** [comment]
@scraly fixed the `key` problem.

**Comment by scraly** [comment]
It's good for me.

Is it possible to pull all the new events please? 
And then I will merge your PR :)

**Comment by jespino** [comment]
@scraly Merged main branch here

**Comment by jespino** [comment]
By the way, after this is merged is going to be easy to add more filters, so I was thinking about a "Online" filter, any other one that you are missing here?

**Comment by scraly** [comment]
Perfect
But sorry I merged two/three new PRs
Can you merged them again here and then I will merge this PR right after

**Comment by jespino** [comment]
@scraly done

**Pull request merged**